packages/ses/README.md

```diff
@@ -285,37 +294,82 @@ const c1 = new Compartment({}, {}, {
+> is an unnecessary performance penalty for the SES runtime.
+> Instead, the SES shim accepts a pre-compiled module source duck-type that
 > is tightly coupled to the shim implementation.
-> Third party modules can provide suitable implementations and even move the
-> compile step to build time instead of runtime.
 
 A compartment can also link a module in another compartment.
-Each compartment has a `module` function that accepts a module specifier
-and returns the module exports namespace for that module.
-The module exports namespace is not useful for inspecting the exports of the
-module until that module has been imported, but it can be passed into the
-module map of another Compartment, creating a link.
 
 ```js
 const c2 = new Compartment({}, {
-  'c1': c1.module('./main.js'),
+  'c1': {
+    source: './main.js',
+    compartment: c1,
+  },
 }, {
   name: "second compartment",
   resolveHook,
   importHook,
 });
 ```
 
-### importHook aliases
+#### Module Descriptors
+
+Comparments can load and initialize module namespaces from module descriptors.
+Like property descriptors, module descriptors are ordinary objects with various
+forms.
+
+##### Descriptors with `source` property
+
+- If fhe value of the `source` property is a string, the parent compartment
+  loads the module but the compartment itself initializes the module.
+
+- Otherwise, if the value of the `source` property is the module source, the
+  module is initialized from the module source.
+
+- Otherwise, the value of the `source` property must be an object. The module
+  is loaded and initialized from the object according to the [virtual module
+  source](#VirtualModuleSource) pattern.
+
+If the `importMeta` property is present, its value must be an object. The
+default `importMeta` object is an empty object.
+
+Compartments copy the `importMeta` object properties into the module
+`import.meta` object like `Object.assign`.
+
+If the `specifier` property is present, its value is coerced into a string and
+becomes the referrer specifier of the module, on which all import specifiers
+are resolved using the `resolveHook`.
+
+##### Descriptors with `namespace` property
+
+- If fhe value of the `namespace` property is a string, the descriptor shares a
+  module to be loaded and initialized by the compartment referred by the
+  `compartment` property.
+
+    - If the `compartment` property is present, its value must be a
+      compartment.
+    - If absent, the `compartment` property defaults to the compartment being
+      constructed in the `modules` option, or being hooked in the `loadHook`
+      and `loadNowHook` options.
+
+- Otherwise, if the value of the `namespace` property is a module namepace, the
+  descriptor shares a module that is already available.
+
+- Otherwise, the value of `namespace` property must be an object. The module is
+  loaded and initialized from the object according to the [virtual module
+  namespace](#VirtualModuleNamespace) pattern.
```

Comment:
What does #VirtualModuleNamespace link to? I don't see it.

Reply:
It’s an internal anchor to the corresponding heading.

Reply:
Where is that corresponding heading? I did not see it.